pkg/asset/deployment.go

```diff
@@ -40,12 +68,19 @@ func (d *deployment) New() *appsv1.Deployment {
 			},
 		},
 		Spec: appsv1.DeploymentSpec{
-			Replicas: &replicas,
+			Replicas: &DefaultReplicas,
 			Selector: &metav1.LabelSelector{
 				MatchLabels: map[string]string{
 					values.SelectorLabelKey: values.SelectorLabelValue,
 				},
 			},
+			Strategy: appsv1.DeploymentStrategy{
+				Type: appsv1.RollingUpdateDeploymentStrategyType,
+				RollingUpdate: &appsv1.RollingUpdateDeployment{
+					MaxUnavailable: &maxUnavailable,
+					MaxSurge:       &maxSurge,
```

Comment:
Maybe we could talk about this in our stand-up tomorrow. I want to make sure I understand what the behavior will be in this situation.

Reply:
We talked about it and Max sent some demos in Slack of the various cases and it looks like it's behaving like we want.